src/go/k8s/controllers/core/redpandacluster_controller.go

```diff
@@ -0,0 +1,417 @@
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/utils/pointer"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	seedSuffix	= "-seed"
+	dataDirectory	= "/var/lib/redpanda/data"
+	fsGroup		= 101
+
+	debugLevel	= 2
+)
+
+// RedpandaClusterReconciler reconciles a RedpandaCluster object
+type RedpandaClusterReconciler struct {
+	client.Client
+	Log	logr.Logger
+	Scheme	*runtime.Scheme
+}
+
+// +kubebuilder:rbac:groups=core.vectorized.io,resources=redpandaclusters,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=core.vectorized.io,resources=redpandaclusters/status,verbs=get;update;patch
+// +kubebuilder:rbac:groups=core.vectorized.io,resources=redpandaclusters/finalizers,verbs=update
+// +kubebuilder:rbac:groups=apps,resources=statefulsets,verbs=get;list;watch;create;update;patch;
+// +kubebuilder:rbac:groups=core,resources=services,verbs=get;list;watch;create;update;patch;
+// +kubebuilder:rbac:groups=core,resources=configmaps,verbs=get;list;watch;create;update;patch;
+// +kubebuilder:rbac:groups=core,resources=pods,verbs=get;list;watch;
+
+// Reconcile is part of the main kubernetes reconciliation loop which aims to
+// move the current state of the cluster closer to the desired state.
+// the RedpandaCluster object against the actual cluster state, and then
+// perform operations to make the cluster state reflect the state specified by
+// the user.
+//
+// For more details, check Reconcile and its Result here:
+// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.7.0/pkg/reconcile
+// nolint:funlen // The complexity of Reconcile function will be address in the next version
+func (r *RedpandaClusterReconciler) Reconcile(
+	ctx context.Context, req ctrl.Request,
+) (ctrl.Result, error) {
+	log := r.Log.WithValues("redpandacluster", req.NamespacedName)
+
+	var redpandaCluster corev1alpha1.RedpandaCluster
+	if err := r.Get(ctx, req.NamespacedName, &redpandaCluster); err != nil {
+		log.Error(err, "unable to fetch RedpandaCluster")
+		// we'll ignore not-found errors, since they can't be fixed by an immediate
+		// requeue (we'll need to wait for a new notification), and we can get them
+		// on deleted requests.
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	var svc corev1.Service
+
+	err := r.Get(ctx, types.NamespacedName{Name: redpandaCluster.Name, Namespace: redpandaCluster.Namespace}, &svc)
+	if err != nil && !errors.IsNotFound(err) {
+		log.V(debugLevel).Info("unable to fetch Service resource")
+		return ctrl.Result{}, err
+	}
+
+	if errors.IsNotFound(err) {
+		log.V(debugLevel).Info("creating headless service")
+
+		if err = r.createHeadlessService(ctx, &redpandaCluster); err != nil {
+			log.Error(err, "Failed to create new service",
+				"Service.Namespace", redpandaCluster.Namespace,
+				"Service.Name", redpandaCluster.Name)
+
+			return ctrl.Result{}, err
+		}
+	}
+
+	var seedConfigMap corev1.ConfigMap
```

Comment:
No. My intention is to distinguish `seed` server (1 pod) from the rest of the replicas. If user set replica to more than 1 in custom resource, then it will create config map with the statful set without `seedSuffix`. 

The difference between `seedConfigMap` and `configMap` would be in the seed server. Using `seedConfigMap` just add seed server list and create the resource in kuberentes api server.